lib/dynamo/dynamo/jobs.py

```diff
@@ -73,6 +84,19 @@ def put_jobs(user_id: str, jobs: list[dict], dry_run: bool = False) -> list[dict
     return prepared_jobs
 
 
+def _get_product_from_archive(job: dict) -> asf.ASFProduct | None:
+    if job.get('job_type') == 'ARIA_S1_GUNW':
+        params = job['job_parameters']
+
+        return aria_s1_gunw.get_product(
+            reference_date=datetime.date.fromisoformat(params['reference_date']),
+            secondary_date=datetime.date.fromisoformat(params['secondary_date']),
```

Comment:
`get_product` accepts ISO-formatted date strings, so no need to convert to `date` object here.